content/blog/understanding-the-dom/index.md

```diff
@@ -0,0 +1,519 @@
+	<li>Bananas</li>
+	<li>Apples</li>
+	<li>Oranges</li>
+</ol>
+```
+
+In fact, the default metadata that is defaulted by specific tags can be directly applied manually to an element of a different tag. The metadata that is passed to the browser when using `<li>` is typically involving that element pertaining to a listitem, using the `role` attribute, we can add that information to a `<div>` itself.
+
+```html
+<ol>
+	<div role="listitem">Bananas</div>
+	<div role="listitem">Apples</div>
+	<div role="listitem">Oranges</div>
+</ol>
+```
+
+> It's worth mentioning that this example is generally considered malpractice. While you may have been able to preserve _some_ of the metadata from an `<li>` tag in a `<div>` element, it's extremely difficult to catch all of the defaults a browser might apply to the original tag that may enhanse a sight-impaired user's experience using screen-readers.
+>
+> This is all to say: unless you have a **really** good reason for using `role` rather than an approprate tag, stick with the related tag. Just as any other form of engineering, properly employing HTML requires nuance and logic to be deployed at the hand of the implementing developer.
+
+# Element Metadata {#interacting-with-elements-using-js}
+
+If you've ever written a website that had back-and-forth communication between HTML and JavaScript, you're likely aware that you can access DOM elements from JavaScript: modifying, reading, and creating them to your heart's content.
+
+Let's look at some of the built-in utilities at our disposal for doing so:
+
+- [The `document` global object](document-global-object)
+- [The `Element` base class](element-class)
+- [The event system](#events)
+
+## Document Global Object {#document-global-object}
+
+[As mentioned before, the DOM tree must contain one root node](#the-dom). This node, for any instance of the DOM, is the document entry point. When in the browser, this entry point is exposed to the developer with [the global object `document`](https://developer.mozilla.org/en-US/docs/Web/API/Document). This object has various methods and properties to assist in a meaningful way. For example, given a standard HTML5 document:
+
+```html
+<!DOCTYPE html>
+<html>
+<head>
+	<title>This is a page title</title>
+</head>
+<body>
+	<p id="mainText">
+    This is the page body
+		<span class="bolded">and it contains</span>
+		a lot of various content within
+    <span class="bolded">the DOM</span>
+	</p>
+</body>
+</html>
+```
+
+The `document` object has the ability to get the `<body>` node ([`document.body`](https://developer.mozilla.org/en-US/docs/Web/API/Document/body)), the `<head>` node ([`document.head`](https://developer.mozilla.org/en-US/docs/Web/API/Document/head)), and even the doctype ([`document.doctype`](https://developer.mozilla.org/en-US/docs/Web/API/Document/doctype)).
+
+![A screenshot of the Chrome debugger console displaying those properties](first_document_properties.png)
+
+### Querying Elements
```

Comment:
I think it may be a good idea to bring up `document.getElement(s)By___()` because `document.getElementById()`:
- is frequently used
- has better browser support than querying
- and is more performant than `document.querySelector("#unique-id")`

Reply:
I don't generally disagree with any of these points, but I think this post is meant to be more focused on concept - not implementation

Reply:
Ahhh I gotcha